front/lib/extract_events.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  cloneBaseConfig,
+  DustProdActionRegistry,
+} from "@app/lib/actions/registry";
+import { runAction } from "@app/lib/actions/server";
+import { getInternalBuilderOwner } from "@app/lib/auth";
+import {
+  getRawExtractEventMarkersFromText,
+  sanitizeRawExtractEventMarkers,
+} from "@app/lib/extract_event_markers";
+import { EventSchema, ExtractedEvent } from "@app/lib/models";
+import logger from "@app/logger/logger";
+import { WorkspaceType } from "@app/types/user";
+
+/**
+ * Gets the markers from the doc and calls _processExtractEvent for each of them
+ */
+export async function processExtractEvents(
+  workspaceId: string,
+  documentId: string,
+  documentText: string
+) {
+  const owner = await getInternalBuilderOwner(workspaceId);
+  const rawMarkers = getRawExtractEventMarkersFromText(documentText);
+  const markers = sanitizeRawExtractEventMarkers(rawMarkers);
+
+  await Promise.all(
+    Object.keys(markers).map(async (marker) => {
+      await _processExtractEvent(
+        owner,
+        marker,
+        markers[marker],
+        documentText,
+        documentId
+      );
+    })
+  );
+}
+
+/**
+ * Gets the schema for the marker, runs the Dust app to extract properties, and saves the event(s)
+ */
+async function _processExtractEvent(
+  owner: WorkspaceType,
+  sanitizedMarker: string,
+  markers: string[],
+  documentText: string,
+  documentId: string
+) {
+  const schema: EventSchema | null = await EventSchema.findOne({
+    where: {
+      workspaceId: owner.id,
+      marker: sanitizedMarker,
+      status: "active",
+    },
+  });
+  if (!schema) {
+    return null;
```

Comment:
super small nit: `return null` here makes the implicit signature of this function a little weird (`void | null`). I think just `return` would make more sense